src/nvme/fabrics.c

```diff
@@ -1113,6 +1122,9 @@ static struct nvmf_discovery_log *nvme_discovery_log(
 			nvme_msg(r, LOG_INFO,
 				 "%s: discover try %d/%d failed, error %d\n",
 				 name, retries, args->max_retries, errno);
+			if (*log_args.result &&
+			    !(*log_args.result & NVME_SC_DNR))
+				continue;
```

Comment:
`continue` in a `do`-`while` loop jumps to the `while` condition: `while (genctr != le64_to_cpu(log->genctr) && ++retries < args->max_retries);`. Is accessing the header valid valid if the command returned an error status? Presumably no data would have been transferred in that case.

Reply:
There were some nasty surprises with this series. I've reverted the whole series for now.